protocol/rpcprovider/rpcprovider_server.go

```diff
@@ -209,7 +210,11 @@ func (rpcps *RPCProviderServer) initRelay(ctx context.Context, request *pairingt
 		}
 	}(relaySession) // lock in the session address
 	// parse the message to extract the cu and chainMessage for sending it
-	chainMessage, err = rpcps.chainParser.ParseMsg(request.RelayData.ApiUrl, request.RelayData.Data, request.RelayData.ConnectionType, request.RelayData.GetMetadata(), 0)
+	extensionInfo := extensionslib.ExtensionInfo{LatestBlock: 0, ExtensionOverride: request.RelayData.Extensions}
```

Comment:
add a comment explaining providers do not parse extensions they rather let the user determine what extensions they want